setuptools/config/_apply_pyprojecttoml.py

```diff
@@ -150,7 +150,9 @@ def _guess_content_type(file: str) -> str | None:
     raise ValueError(f"Undefined content type for {file}, {msg}")
 
 
-def _long_description(dist: Distribution, val: _ProjectReadmeValue, root_dir: StrPath):
+def _long_description(
+    dist: Distribution, val: _ProjectReadmeValue, root_dir: StrPath | None
```

Comment:
Similar comment to this and all the other functions: I think `root_dir` is always StrPath.